py4DSTEM/process/fit/fit.py

```diff
@@ -46,27 +54,39 @@ def fit_2D(function, data, data_mask=None, popt_guess=None):
                     same shape as data, containing the fit values
     """
     shape = data.shape
+    shape1D = [1,np.prod(shape)]
     x,y = np.arange(shape[0]),np.arange(shape[1])
+    x,y = np.linspace(0, 1, shape[0]),np.linspace(0, 1, shape[1])
     ry,rx = np.meshgrid(y,x)
     rx_1D = rx.reshape((1,np.prod(shape)))
     ry_1D = ry.reshape((1,np.prod(shape)))
     xy = np.vstack((rx_1D, ry_1D))
 
-    if data_mask is not None:
-        rx_1D_known = rx_1D[data_mask.reshape((1,np.prod(shape)))]
-        ry_1D_known = ry_1D[data_mask.reshape((1,np.prod(shape)))]
-        xy_known = np.vstack((rx_1D_known,ry_1D_known))
-        data_1D = data[data_mask]
-        if popt_guess is not None:
-            popt, pcov = curve_fit(function, xy_known, data_1D, p0=popt_guess)
+    # if robust fitting is turned off, set number of robust iterations to 0
+    if robust==False:
+        robust_steps=0
+
+    # least squares fitting - 1st iteration 
+    for k in range(robust_steps+1):
+        if k == 0:
+            if popt_guess is None:
+                popt = np.zeros((1,len(signature(function).parameters)-1))
```

Comment:
add a line with else: popt_guess = popt, OR fix the naming so there's one variable

Reply:
popt_guess has been renamed to just popt everywhere.